### PR TITLE
Use restartRequired condition when volumes cannot be updated

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
Currently, if the volumes cannot be live updated we
are setting the `VirtualMachineInstanceVolumesChange` condition
to the vm.
This is wrong for two reasons:
- `VirtualMachineInstanceVolumesChange` is a vmi condition and not a vm condition.
- We need to specify that the vm requires a restart if validateVolume fails

### Release note

```release-note
NONE
```

/cc @alicefr @vladikr @mhenriks

### CODE REVIEW

**Comment by alicefr** [comment]
Looks good, many thanks for catching this!
/approve

**Comment by acardace** [comment]
@fossedihelm @alicefr shouldn't we have a unit test catching this?

**Comment by fossedihelm** [comment]
/hold
Found a bug in the unit test

**Comment by fossedihelm** [comment]
@acardace Pushed unit test and a mini fix. PTAL

File: pkg/virt-controller/watch/vm_test.go [hunk 1]
```diff
@@ -6155,6 +6155,52 @@ var _ = Describe("VirtualMachine", func() {
...
+										virtconfig.VolumeMigration,
+									},
+								},
+							},
+						},
+					})
+					vm, vmi := DefaultVirtualMachine(true)
+					vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+						Name: "vol1"})
+					vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
```
Comment (alicefr) [inline pkg/virt-controller/watch/vm_test.go:6177]: nit: you can merge the 2 volumes into a single append

Comment (fossedihelm) [reply pkg/virt-controller/watch/vm_test.go:6177]: :man_facepalming:

**Pull request merged**